modules/users/user.nix

```diff
@@ -65,10 +73,17 @@
     };
 
     shell = mkOption {
-      type = types.either types.shellPackage types.path;
-      default = "/sbin/nologin";
+      type = types.nullOr (types.either types.shellPackage types.path);
+      default = "/usr/bin/false";
```

Comment:
You can break your shell by starting to manage your user with nix-darwin unless you explicitly `null` this out, right? I was thinking we could just make `null` create users with `/usr/bin/false` but otherwise act the same way as the default. Not sure if that’s a good idea or not.

Reply:
This is a good catch, I think this bug has been around since #818 and I'm not sure if anyone has been hit by it yet as it's currently gated behind `users.knownUsers` and my suspicion is that most users only add themselves to `users.knownUsers` when they want to update their shell as most other options are less useful.

I'll add the change to make `null` create users with `/usr/bin/false` and but still mean don't update this user's shell. This change should be superseded by adding a `type` option because then we could not pass the `-shell` flag to `sysadminctl` and it would set the correct shell based on whether `-roleAccount` was passed or not (alternatively we could determine the default passed on the `type` in nix-darwin as well)